src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/NumberPayloadGeneratorAdapterUIHandler.java

```diff
@@ -0,0 +1,306 @@
+                            .addComponent(fromField)
+                            .addComponent(toField)
+                            .addComponent(stepField)
+                            .addGroup(
+                                    layout.createSequentialGroup()
+                                    .addComponent(getPayloadsPreviewGenerateButton())
+                                    .addComponent(getSaveButton()))
+                            .addComponent(payloadsPreviewScrollPane)));
+            layout.setVerticalGroup(layout.createSequentialGroup()
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                            .addComponent(fromLabel)
+                            .addComponent(fromField))
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                            .addComponent(toLabel)
+                            .addComponent(toField))
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                            .addComponent(stepLabel)
+                            .addComponent(stepField))
+                    .addGroup(
+                            layout.createParallelGroup()
+                            .addComponent(getPayloadsPreviewGenerateButton())
+                            .addComponent(getSaveButton()))
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                            .addComponent(payloadsPreviewLabel)
+                            .addComponent(payloadsPreviewScrollPane)));
+
+        }
+
+        @Override
+        protected NumberPayloadGenerator getPayloadGenerator() {
+            if (!validate()) {
+                return null;
+            }
+            long f = Long.parseLong(fromField.getText());
+            long t = Long.parseLong(toField.getText());
+            long s = Long.parseLong(stepField.getText());
+            return new NumberPayloadGenerator(f, t, s);
+        }
+
+        @Override
+        public void init(MessageLocation messageLocation) {
+        }
+
+        @Override
+        public JPanel getComponent() {
+            return fieldsPanel;
+        }
+
+        @Override
+        public void setPayloadGeneratorUI(NumberPayloadGeneratorUI payloadGeneratorUI) {
+            oldGenerator = payloadGeneratorUI;
+            fromField.setText(String.valueOf(payloadGeneratorUI.getPayloadGenerator().getFrom()));
+            toField.setText(String.valueOf(payloadGeneratorUI.getPayloadGenerator().getTo()));
+            stepField.setText(String.valueOf(payloadGeneratorUI.getPayloadGenerator().getStep()));
+        }
+
+        @Override
+        public NumberPayloadGeneratorUI getPayloadGeneratorUI() {
+            if (oldGenerator != null) {
```

Comment:
It should return the old generator only if there was no changes to the values of the generator (the check should be done when validating and the oldGenerator set to null).